libs/estdlib/src/erlang.erl

```diff
@@ -651,6 +652,16 @@ binary_to_atom(_Binary, _Encoding) ->
 binary_to_integer(_Binary) ->
     erlang:nif_error(undefined).
 
+%%-----------------------------------------------------------------------------
+%% @param   Binary  Binary to parse for integer
+%% @returns the integer represented by the binary
+%% @doc     Parse the text in a given binary as an integer.
+%% @end
+%%-----------------------------------------------------------------------------
+-spec binary_to_integer(Binary :: binary(), Base :: 2..36) -> integer().
+binary_to_integer(_Binary, Base) when 2 =< Base, 36 >= Base ->
```

Comment:
The test for 2..36 is already done in C, we can save few bytes by not doing it here.